docs/testing_image.md

```diff
@@ -0,0 +1,254 @@
+# Testing Job Image Overview
+
+This document provides additional detail about
+the images described in the [canonical testing] proposal.
+
+The basic idea is that each testing job will map to one of a handful of job
+classes: unit, integration, verify, e2e, node, etc.
+
+Each class of test job will define an image. The image includes:
+
+* all the necessary dependencies to complete the job.
+  - Required packages.
+  - Scripts/binaries that drive the testing.
+  - Common arguments and environment variables
+* logic to deal with parameters that vary between jobs of the same class.
+  - Arguments and environment variables that make each job unique
+* logic to deal with parameters that change between users.
+  - Arguments and environment variables that are unique for each user
+    (credentials, for example).
+* logic to deal with parameters that configure what happens to results
+  - Credentials/location to upload results.
+
+This will help make sure running these test is easy and consistent.
+
+## Common logic
+
+Each job requires logic to perform some common behavior:
+
+* Find credentials in a known location
+* Understand which job name, commit and optionally PR is under test.
+* Copy and extract a tarball of the source and/or binaries from a known prefix.
```

Comment:
may want to clarify how this actually happens - do the images download from git? do we map in a volume? do we just download build artifacts from GCS?